resources/ui/src/components/WebSocketManager.jsx

```diff
@@ -0,0 +1,198 @@
+import { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import useWebSocket from 'react-use-websocket';
+import { DynamoDBClient, UpdateItemCommand } from '@aws-sdk/client-dynamodb';
+import React from 'react';
+import {
+  withAuthenticator,
+} from '@aws-amplify/ui-react';
+import { fetchAuthSession } from 'aws-amplify/auth';
+
+import Modal from "@cloudscape-design/components/modal";
+import Box from "@cloudscape-design/components/box";
+import SpaceBetween from "@cloudscape-design/components/space-between";
+import Button from "@cloudscape-design/components/button";
+
+function WebSocketManager({ user, websocketURL, websocketStateTable, region, toast }) {
+  const [connectionId, setConnectionId] = useState("");
+  const [creds, setCreds] = useState({});
+  const [hasErrored, setHasErrored] = useState(false);
+
+
+  // Getting STS credentials for user
+  useEffect(() => {
+    const getSession = async () => {
+      try {
+        const { credentials, identityId, tokens } = await fetchAuthSession();
+        setCreds({
+          ...credentials,
+          ...tokens,
+          identityId
+        });
+      } catch (error) {
+        console.error("Error fetching session: ", error);
+      }
+    };
+    getSession();
+  }, [user]);
+
+    // check if this component has actually unmounted
+    useEffect(() => {
+      return () => {
+        didUnmount.current = true;
+      };
+    }, []);
+
+  const didUnmount = useRef(false);
+
+  const { sendMessage, lastMessage, readyState } = useWebSocket(
+    `${websocketURL}?idToken=${creds.idToken}`,
+    {
+      onOpen: (event) => {
+        console.log("Connected to websocket", event);
+        sendMessage(JSON.stringify({ action: "whoami" }))
+        setHasErrored(false)
+      },
+      onError: (event) => console.log(event) || creds.idToken? setHasErrored(true) : null,
```

Comment:
this is a number of commits behind I am afraid. I have changed this logic to handle multiple error reasons.